pele_platform/Adaptive/simulation.py

```diff
@@ -145,62 +218,97 @@ def run_adaptive(args: pv.EnviroBuilder) -> pv.EnviroBuilder:
+            env.solvent, env.obc_tmp, env.template_folder, env.obc_file, env.logger
+        )
         solvent.generate()
-        
+
         # Build PCA
         if env.pca_traj:
             pca_obj = pca.PCA(env.pca_traj, env.pele_dir)
             env.pca = pca_obj.generate(env.logger)
-        
+
         # Core constraints based on SMILES string
         if env.constrain_core:
-            smiles_input_pdb = os.path.join(env.pele_dir, env.adap_ex_input.split(",")[0])
-            smiles = smi.SmilesConstraints(smiles_input_pdb, env.constrain_core, env.residue, env.chain, env.constrain_core_spring)
+            smiles_input_pdb = os.path.join(
+                env.inputs_dir, env.adap_ex_input.split(",")[0]
+            )
+            smiles = smi.SmilesConstraints(
+                smiles_input_pdb,
+                env.constrain_core,
+                env.residue,
+                env.chain,
+                env.constrain_core_spring,
+            )
             smi_constraint = smiles.run()
-            env.constraints = env.constraints[0:1] + smi_constraint + env.constraints[1:]
-        
-        ####### Add waters, if needed
-        input_waters = [input.strip().strip('"') for input in env.adap_ex_input.split(",")]
-        input_waters = [os.path.join(env.pele_dir, inp) for inp in input_waters]
-        water_obj = wt.WaterIncluder(input_waters, env.n_waters, 
-            user_waters=env.waters, ligand_perturbation_params=env.parameters, 
-            water_center=args.water_center, water_radius=env.water_radius,
-            allow_empty_selectors=env.allow_empty_selectors, water_temp=env.water_temp,
-            water_trials=env.water_trials, water_overlap=env.water_overlap,
-            water_constr=env.water_constr, test=env.test, water_freq=env.water_freq,
-            ligand_residue=env.residue)
+            env.constraints = (
+                env.constraints[0:1] + smi_constraint + env.constraints[1:]
+            )
+
+        # Waters
+        input_waters = [
+            input.strip().strip('"') for input in env.adap_ex_input.split(",")
+        ]
+        input_waters = [os.path.join(env.inputs_dir, inp) for inp in input_waters]
+        water_obj = wt.WaterIncluder(
+            input_waters,
+            env.n_waters,
+            user_waters=env.waters,
+            ligand_perturbation_params=env.parameters,
+            water_center=args.water_center,
+            water_radius=env.water_radius,
+            allow_empty_selectors=env.allow_empty_selectors,
+            water_temp=env.water_temp,
+            water_trials=env.water_trials,
+            water_overlap=env.water_overlap,
+            water_constr=env.water_constr,
+            test=env.test,
+            water_freq=env.water_freq,
+            ligand_residue=env.residue,
+        )
         water_obj.run()
         env.parameters = water_obj.ligand_perturbation_params
 
-        #metrics
+        # Metrics builder - RMSD PDB, atom_dist
```

Comment:
Since you are improving this comment, maybe it is worth making it more self-explanatory.